### PR TITLE
Create .lgtm.yml

### PR DESCRIPTION
**Pull request by XVilka**
Trying to address https://github.com/neovim/neovim/issues/11364

### CODE REVIEW

**Comment by justinmk** [comment]
Installed LGTM in the GitHub "Neovim" org. Maybe re-push this to trigger it?

File: .lgtm.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+extraction:
+    cpp:
+        after_prepare:
+            - export GNU_MAKE=make
+        index:
+            build_command:
+                - $GNU_MAKE -j2 -s
```
Comment (blueyed) [inline .lgtm.yml]: Please use an indentation of 2 spaces when re-pushing.
(at least that's why I assume the convention here is)

**Comment by blueyed** [comment]
Got triggered in https://github.com/neovim/neovim/pull/11477 already.

**Comment by XVilka** [comment]
See https://help.semmle.com/lgtm-enterprise/user/help/lgtm.yml-configuration-file.html for more configuration details

File: .lgtm.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+extraction:
+  cpp:
+    after_prepare:
+      - export GNU_MAKE=make
+    index:
+      build_command:
+        - $GNU_MAKE -j2 -s
```
Comment (justinmk) [inline .lgtm.yml]: `ninja` is used on travis CI, I wonder if there's a way to use it in LGTM (would need to be installed).

**Comment by justinmk** [comment]
why is this needed? seems like LGTM is already analyzing somehow:
https://lgtm.com/projects/g/neovim/neovim/alerts/?mode=list

**Comment by XVilka** [comment]
I think it recognized the default build mode.

**Comment by justinmk** [comment]
@XVilka another question: is there a way to disable the PR comments from LGTM? OTOH, LGTM reports "success" in the GitHub "checks" table, even if a new issue was found.

**Comment by XVilka** [comment]
OK, JavaScript analysis is now disabled. Thanks to the LGTM staff! 
@justinmk yes, it is possible - see https://lgtm.com/help/lgtm/managing-automated-code-review

**Comment by justinmk** [comment]
> If there is a new alert it shows in the check status: [scikit-learn/scikit-learn#12519 (comment)](https://github.com/scikit-learn/scikit-learn/issues/12519#issuecomment-436071327)

yes, but as mentioned there (and above):

> Yes, we now need to check the LGTM status manually, but I guess it's worthwhile, since the comments from LGTM seems annoying.

bleh :)

File: .lgtm.yml (hunk 2 above)
Comment (justinmk) [reply .lgtm.yml]: is `ninja` available?

**Comment by XVilka** [comment]
Yes, ninja is available.

**Comment by justinmk** [comment]
That might make it a bit faster, though it seems like this change saves 2+ minutes already. Will you change it to use ninja here?

File: .lgtm.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+extraction:
+  cpp:
+    after_prepare:
+      - export GNU_MAKE=make
+      - export BUILD_TYPE=Ninja
+    index:
+      build_command:
+        - $GNU_MAKE -j2 -s
```
Comment (justinmk) [inline .lgtm.yml]: `-j2` isn't needed with ninja. Also not sure if `-s` is wanted

**Comment by justinmk** [comment]
Doesn't look like this improved the build time (still 18 min). Also confused about why `GNU_MAKE` is still there.

Seems like this file isn't actually needed.

**Pull request closed without merging**